### ISSUE TITLE
Pivoting a very wide lazy table throws a c-stack error.

### ISSUE THREAD

**Issue by abalter** [tidyverse/dbplyr#1217]
I created a wide table: >100 columns. As a tibble I can pivot it to three columns. As a memdb, SQLite in-memory table, or arrow-->duckdb table I get a c-stack error:

>Error: C stack usage  7972212 is too close to the limit

When I reduce the number of columns to , say 80, I don't get the error.

I created a reprex, but [for some reason it won't display that error](https://github.com/tidyverse/reprex/issues/438). However, I am including the code below:

```r
library(tidyverse)
library(dbplyr)
library(RSQLite)
library(arrow)

Nids = 10
Nyears = 10
Ndates = 12*Nyears
start_year = 2010

"*******  Create Very Wide Table  *********"
tb_wide =
  crossing(
    id = str_glue("id_{1:Nids}"),
    year = start_year:(start_year+Nyears-1),
    month = month.abb %>% tolower()
  ) %>%
  tibble() %>%
  mutate(value = rnorm(Nids*Ndates)) %>%
  unite(month, year, col="date", sep="_") %>%
  pivot_wider(names_from=date, values_from = value)

tb_wide %>% nrow()
tb_wide %>% ncol()
colnames(tb_wide)

"******  Create lazy arrow table  *****"
readr::write_csv(tb_wide, "tb_wide.csv")
tb_arrow = read_csv_arrow("tb_wide.csv", as_data_frame = F)
tb_arrow %>% nrow()
tb_arrow %>% ncol()

"******  Create memdb table  ******"
tb_mdb = memdb_frame(tb_wide)
tb_arrow %>% nrow()
tb_arrow %>% ncol()

"******  Create in-memory sqlite table  ******"
con = dbConnect(RSQLite::SQLite(),":memory")
copy_to(con, tb_wide, "tb_wide")
dbListTables(con)
tb_db = tbl(con, "tb_wide")
tb_arrow %>% nrow()
tb_arrow %>% ncol()


"******  Try Pivoting  ******"
tb_long =
  tb_wide %>%
  pivot_longer(cols=-id, names_to="date", values_to = "value")

dim(tb_long)
colnames(tb_long)

tb_arrow %>%
  to_duckdb() %>%
  pivot_longer(cols=-id, names_to="date", values_to = "value")

tb_mdb %>%
  pivot_longer(cols=-id, names_to="date", values_to = "value")

tb_db %>%
  pivot_longer(cols=-id, names_to="date", values_to = "value")

print(nonexistent_variable)

sessionInfo()

```

**Comment by mgirlich** [comment]
I can't reproduce this but I think this has to do with the many `union_all()` it does in the background. The data structure has to be changed for this in order to make this work correctly.

**Comment by mgirlich** [comment]
Yes, I don't get an error, not even with `Nyears = 24`. You can try using the dev version of dblyr. Otherwise, this can also be related to the system you're using.

**Comment by rkb965** [comment]
I also get this C stack error. My personal use case was with duckdb and pivot_longer, 500 columns. The query ran smoothly as a tibble but throws a C stack error with duckdb (works well with 10 columns, haven't yet explored the limit).  

I also get the C stack error with the above repex. Any idea what system details might be relevant to this?

**Comment by mgirlich** [comment]
You could try with the dev version of dbplyr. Otherwise, this has to wait until a series of `union()` resp. `union_all()` is handled differently. It is on the list of things I want to tackle but I don't know yet when I have time for this.

**Comment by mgirlich** [comment]
The dev version now handles a sequence of `union()` better. Can you install it via `devtools::install_github("tidyverse/dbplyr")` and give feedback whether this solves your issues?

**Comment by abalter** [comment]
Will do!

**Comment by mgirlich** [comment]
Thanks for the feedback.
Closed by #1270.
